tests/infra/github.py

```diff
@@ -112,8 +112,25 @@ def __init__(self):
             if "heads/release" in branch
         ]
 
+    def _filter_released_tags(self, tags):
+        # From a list of tags ordered by semver (latest first), filter out the ones
+        # that don't have a release yet
+        first_release_tag_idx = -1
+        for i, t in enumerate(tags):
+            if not has_release_for_tag_name(t):
+                LOG.debug(f"No release available for tag {t}")
+                first_release_tag_idx = i
+            else:
+                break
+
+        return tags[first_release_tag_idx + 1 :]
+
     def get_latest_dev_tag(self):
-        return self.tags[-1]
```

Comment:
Is this called once per CI? If so, it sounds like another API throttling issue waiting to happen.

Reply:
We no longer call the GitHub API (at all) but rely on the local git checkout instead.